distribution/src/bin-filemode-755/hz-start

```diff
@@ -37,9 +37,18 @@ fi
 # HAZELCAST_CONFIG holds the configuration path.
 # If the the path is relative, it is relative to the Hazelcast installation directory (HAZELCAST_HOME).
 # The path can be absolute.
-# By default, it is set to $HAZELCAST_HOME/config/hazelcast.xml
+# By default, it is set to $HAZELCAST_HOME/config/hazelcast.xml, with fallback to $HAZELCAST_HOME/config/hazelcast.yaml
+# if the xml file is not present.
 if [ -z "$HAZELCAST_CONFIG" ]; then
     HAZELCAST_CONFIG="config/hazelcast.xml"
+    if [ ! -f "$HAZELCAST_HOME/$HAZELCAST_CONFIG" ]; then
+        HAZELCAST_CONFIG="config/hazelcast.yaml"
+    fi
+
+    if [ ! -f "$HAZELCAST_HOME/$HAZELCAST_CONFIG" ]; then
+        echo "Configuration file is missing. Create hazelcast.xml or hazelcast.yaml in $HAZELCAST_HOME/config or set the HAZELCAST_CONFIG environment variable."
```

Comment:
I'm not sure if we should exit. According to https://docs.hazelcast.com/hazelcast/5.0/configuration/checking-configuration we should check for ".yml" as well and if none is present then load the default configuration

Reply:
Is there a reason why we should have different behavior on server distribution? I could imagine it could be surprising for customers using embedded versions while migrating to the server distribution. Also, it brings additional effort on documentation

Reply:
I don't think it makes sense for the distribution to try the options done for embedded:
- classpath - do we want to encourage users to put config inside custom jars? I don't think so.
- working directory - currently that's HAZELCAST_HOME if you run hz-start, but we don't have it documented
- the default config in hazelcast jar is not suitable for server distribution (e.g. it opens on all interfaces, instead of just loopback)

Reply:
fair enough